src/dtype.rs

```diff
@@ -43,6 +47,13 @@ impl PyArrayDescr {
         self.as_ptr() as _
     }
 
+    /// Returns `self` as `*mut PyArray_Descr` and transfers the ownership.
+    ///
+    /// Useful in cases where the descriptor is stolen by the API.
+    pub fn into_dtype_ptr(&self) -> *mut PyArray_Descr {
```

Comment:
But wouldn't it be more helpful if this method were to take `self` by value then?

Reply:
Ah sorry, from the cover it seems that this does not really "transfer ownership", it just increments the reference count. Could you modify the comment in that direction? Thanks!

Reply:
I guess it's really meant to be `<&'a PyArrayDescr>::into_dtype_ptr()`, but that would be a bit too obscure.

Good point - I'll update the docstring.